### ISSUE TITLE
TypeError: 'NoteType' object is not subscriptable when authenticating with 2FA (v3.0.11)

### ISSUE THREAD

**Issue by jezzaaa** [costastf/locationsharinglib#54]
Trying to get the new (v3.0.11) version working, to create a cookie. I run:

```
username='redacted@gmail.com'
password='********'
cookiesfile="MY_COOKIES.email@gmail.com"
from locationsharinglib import CookieGetter
cg = CookieGetter(username, password, cookies_file=cookiesfile)
```

I get the result:

```
Traceback (most recent call last):
  File "test-get-cookies.py", line 7, in <module>
    cg = CookieGetter(username, password, cookies_file=cookiesfile)
  File "/home/jpl/.virtualenvs/locationsharinglib/lib/python3.5/site-packages/locationsharinglib/locationsharinglib.py", line 359, in __init__
    cookies_file=cookies_file)
  File "/home/jpl/.virtualenvs/locationsharinglib/lib/python3.5/site-packages/locationsharinglib/locationsharinglib.py", line 225, in __init__
    self._authenticate()
  File "/home/jpl/.virtualenvs/locationsharinglib/lib/python3.5/site-packages/locationsharinglib/locationsharinglib.py", line 263, in _authenticate
    self._submit_password()
  File "/home/jpl/.virtualenvs/locationsharinglib/lib/python3.5/site-packages/locationsharinglib/locationsharinglib.py", line 310, in _submit_password
    self._handle_prompt(body)
  File "/home/jpl/.virtualenvs/locationsharinglib/lib/python3.5/site-packages/locationsharinglib/locationsharinglib.py", line 318, in _handle_prompt
    data_key = data[0][10][0][0][23]['5004'][12]
TypeError: 'NoneType' object is not subscriptable
```

If I hack _handle_prompt() to print out the value of "data" immediately after json decoding, it shows "None" in all of the arrays in the list.

This is what it looks like:

`[['gf.sicr', None, None, 3, None, None, None, None, None, None, [[[None, None, 'TWO_STEP_VERIFICATION', 5, None, None, None, None, 4, None, 1, 1, 1, None, None, 'Firstname Lastname', 'redacted@gmail.com', 'https://lh3.googleusercontent.com/-cpXahc_wJtY/AAAAAAAAAAI/AAAAAAAAAAA/randomrandomrandom/mo/photo.jpg', None, None, None, None, [], None, 8, {'1004': [None, None, None, 'myaccount.google.com/webapproval', 1, [], [], []]}], [None, None, 'TWO_STEP_VERIFICATION', 4, None, None, None, None, 5, None, 1, 1, 1, None, None, 'Firstname Lastname', 'redacted@gmail.com', 'https://lh3.googleusercontent.com/-cpXahc_wJtY/AAAAAAAAAAI/AAAAAAAAAAA/randomrandomrandom/mo/photo.jpg', None, None, None, None, [], None, 8, {'1005': [[], [], None, 0]}], [None, None, 'TWO_STEP_VERIFICATION', 2, None, None, None, None, 6, None, 1, 1, 1, None, None, 'Firstname Lastname', 'redacted@gmail.com', 'https://lh3.googleusercontent.com/-cpXahc_wJtY/AAAAAAAAAAI/AAAAAAAAAAA/randomrandomrandom/mo/photo.jpg', None, None, None, None, [], {'1006': ['ANDROID']}], [None, None, 'TWO_STEP_VERIFICATION', 3, None, None, None, None, 9, None, 1, 1, 1, None, None, 'Firstname Lastname', 'redacted@gmail.com', 'https://lh3.googleusercontent.com/-cpXahc_wJtY/AAAAAAAAAAI/AAAAAAAAAAA/randomrandomrandom/mo/photo.jpg', None, None, None, None, [], {'1009': ['  48', 'SMS', ['SMS'], None, 'TWO_STEP_PREFERRED']}]], 1, 2, 2, None, 2, 1, None, [[8], None, None, None, 1], [None, None, None, None, 0]]], ['gf.ttu', 0, 'AHnYQLzhB-RandomRandomRandomRandom-ZbMEtKj8uWOkHKRNy-oG3WW3hU4FG2yRGmRJpyS0bnpNk7JNiMSsLAd2L1NThalG2e4TIrDGBHCVfkF7nCk_wOBFftTpP_DCkd-wuiQ-mBx0FBindfMVYT0oRcmWuU']]`

I know that the first factor is working, because if I put in a bogus password, I get a different response (locationsharinglib.locationsharinglibexceptions.InvalidCredentials). I don't get any second factor prompt on my phone.

The "Usage" doco doesn't provide enough detail for me to follow, I'm afraid. And I'm not sure whether I'm doing it wrong, or if there's a change to the Google API that's breaking the lib. I'd be grateful for help in getting this working.

I assume step 1 is to create a valid cookie file, and then step 2 is to use this to get some real work done.

**Comment by costastf** [comment]
Apologies for not updating the documentation or breing clear enough. 3.0.11 removed cookie getter functionality by using a patch that handles 2fa in the main object. You should not need to use cookie getter but pass the credentials to Authenticator that should handle it for you.

**Comment by costastf** [comment]
That should be 
```
username='redacted@gmail.com'
password='********'
cookiesfile="MY_COOKIES.email@gmail.com"
from locationsharinglib import Authenticator
service = Authenticator(username, password, cookies_file=cookiesfile)
```

**Comment by jezzaaa** [comment]
Thanks for that. It makes it clearer. But I'm still not quite there...

I had to manually install pytz (into my virtualenv), perhaps needs to be added to dependencies (I'm not familiar with python libs), but all good now for me.

But now I get the following, and I'm stuck:

```
$ python test.py
Traceback (most recent call last):
  File "test.py", line 14, in <module>
    from locationsharinglib import Authenticator
ImportError: cannot import name 'Authenticator'
```
I can't work out if this is just a newby python error on my part, or some issue with the export in the library. If I import Service instead of Authenticator, I get the previous run-time error, suggesting that Authenticator is not exported.

On a hunch, I updated `__init__.py` and added `Authenticator `to the line "`from .locationsharinglib import Service, ...`". Now it compiles, but I'm back to the same error as before:

`TypeError: 'NoneType' object is not subscriptable`

Sorry if this is a silly error on my part.

**Comment by costastf** [comment]
Sorry I think that should be ```username='redacted@gmail.com'
password='********'
cookiesfile="MY_COOKIES.email@gmail.com"
from locationsharinglib import Service
service = Service(username, password, cookies_file=cookiesfile)```

**Comment by jezzaaa** [comment]
No, that didn't help. I get the same 'TypeError bla bla not subscriptable' error. Where else can I look?

**Comment by costastf** [comment]
Hi @jezzaaa  apologies for the late reply. If things do not work for you in the current release, it means that you are using yet another auth mechanism of Google. Chasing that thing around is not working great but I thought that @lufton 's patch worked for everybody so I parked my investment on making another process to get the cookies interactively. It seems that I have to rekindle that effort... Sadly I am extremely busy for this to move fast. I can provide some demo code soon enough for some people to move forward but nothing really robust or finished.

**Comment by jezzaaa** [comment]
@costastf - no problems. I'm happy with the possibility that it might work for me one day. You've done amazing work so far, which works well for most.

**Comment by costastf** [comment]
Hi all. Please use the manual work around mentioned in https://github.com/costastf/locationsharinglib/issues/50#issuecomment-448939141 to create your cookie file and provide that to the library. I will not be trying to fix this internally as this is a cat and mouse game and I don't have the time to invest to chase behind google everytime they change things. On the other hand I do not think that anyone else has that time either, so I will be closing this issue with the work around mentioned above and will be looking to remove the authentication part from the library as soon as possible and build something that will help with the retrieval of the cookies from a proper interactive browser session. Please be a bit patient on that and use the workaround until then.

**Comment by costastf** [comment]
I have removed the authentication part from locationsharinglib and uploaded version 4 with those changes.
I have made a proof of concept project that drives a browser via selenium and saves the cookies in a compatible with locationsharinglib format. If anyone wants to try out the POC and very crud version because they are feeling adventurous you can follow the documentation at the end of the page https://github.com/costastf/mapscookiegettercli/blob/initial-usability/USAGE.rst .
This is heavily a work in progress and only works for chrome currently, tested only on Mac (but i guess linux should also work) so please do not start creating issues just yet, unless you also want to contribute. I am planing to clean it up and and make it nicer during the upcoming weekends.
I am hoping to have a somewhat properly working version out after the coming weekend supporting chrome for all latest version of Mac, Linux and Windows. 
Of course if anyone is interested in helping out, all help is welcome!

**Comment by costastf** [comment]
Hi all, I have created an alpha of a cookiegetter project that drives a browser through selenium and saves the cookies. Please look at https://github.com/costastf/mapscookiegettercli/blob/master/USAGE.rst for usage (it is quite simple :) ) and report issues at the tracker.
